sympy/solvers/tests/test_solvers.py

```diff
@@ -114,6 +114,446 @@ def test_guess_transcendental():
     assert guess_solve_strategy(a*x**b - y, x)  # == GS_TRANSCENDENTAL
 
 
+def test_solve_io():
+    lx = x
+    nx = x**2 - 4
+    lxly = x - y + 1
+    lxny = x + y**2 - 1
+    nxly = y + x**2 - 1
+    nxny = y**2 + x**2 - 1
+    flags = {}, dict(dict=True), dict(set=True)
+    e = lx
```

Comment:
This test is quite hard to follow. I think that redefining `e` like this makes it harder. Why not just use `lx` below?

Also I think it can probably be made more compact. At leat when dict or set is True the output should be more predictable so we can compress those cases together into something like:
```python
eqs_sol = [
    # eq, sol, solx, soly, solxy, solyx
    (x, [{x:0}], [{x:0}], [], [{x:0}], [{x:0}]),
    (x - y + 1, [{x:y-1}], [{x:y-1}], [{y:x+1}], [{x:y-1}], [{y:x+1}]),
    ...
]
for eq, sol, solx, soly, solxy, solyx in eqs_sol:
    for eqi in [eq, [eq], Eq(eq, 0), [Eq(eq, 0)]:
        assert solve(eq, dict=True) == sol
        assert solve(eq, x, dict=True) == solx
        assert solve(eq, y, dict=True) == soly
        ...
```
That way it's easier to look at `eqs_sol` to see how the different cases compare.

Reply:
Please check new format.